dtfs-central-api/src/v1/controllers/portal/gef-deal/put-gef-deal.status.controller.js

```diff
@@ -1,16 +1,36 @@
 const { validateAuditDetails, generateAuditDatabaseRecordFromAuditDetails } = require('@ukef/dtfs2-common/change-stream');
-const { MONGO_DB_COLLECTIONS, InvalidAuditDetailsError } = require('@ukef/dtfs2-common');
+const { MONGO_DB_COLLECTIONS, DealNotFoundError, ApiError, InvalidPayloadError } = require('@ukef/dtfs2-common');
 const { ObjectId } = require('mongodb');
 const { findOneDeal } = require('./get-gef-deal.controller');
 const { mongoDbClient: db } = require('../../../../drivers/db-client');
 
-const updateDealStatus = async ({ dealId, previousStatus, newStatus, auditDetails }) => {
+/**
+ * Update the status on a GEF deal
+ * @param updateGefDealStatusRequest
+ * @param {string} updateGefDealStatusRequest.dealId - the dealId
+ * @param {import('@ukef/dtfs2-common').DealStatus} updateGefDealStatusRequest.status - the updated status to set
+ * @param {import('@ukef/dtfs2-common').AuditDetails} updateGefDealStatusRequest.auditDetails - the users audit details
+ * @returns {Promise<('@ukef/dtfs2-common').Deal>}
+ */
+const updateGefDealStatus = async ({ dealId, status, auditDetails }) => {
+  const existingDeal = await findOneDeal(dealId);
+
+  if (!existingDeal) {
+    throw new DealNotFoundError(dealId);
+  }
+
+  if (existingDeal.status === status) {
```

Comment:
I think `newStatus` is clearer than `status`, what's the reason for using `status` instead?

Reply:
I was doing this to make the two update deal status controllers identical - I went for `status` to make the `dealUpdate` neater but I'll change them both to `newStatus`

```
  const dealUpdate = {
    previousStatus,
    status,
    updatedAt: Date.now(),
  };
```